src/include/sof/audio/module_adapter/intel/syntax_sugar.h

```diff
@@ -0,0 +1,248 @@
+#endif
+
+#define __in_section(a, b, c) ___in_section(a, b, c)
+
+#ifndef __in_section_unique
+#define __in_section_unique(seg) ___in_section(seg, __FILE__, __COUNTER__)
+#endif
+
+#ifndef __aligned
+#define __aligned(x) __attribute__((__aligned__(x)))
+#endif
+
+#define SECTIONATT(sect_name) __attribute__((section(sect_name)))
+#define BSS_SECTION SECTIONATT(".bss")
+
+/********************************************************************
+ *             COMPILER DIRECTIVE MACROS
+ *********************************************************************/
+#define LIKELY(condition) __builtin_expect((condition), 1)
+#define UNLIKELY(condition) __builtin_expect((condition), 0)
+
+#ifndef STRINGIFY
+#define _STRINGIFY(x) #x
+#define STRINGIFY(s) _STRINGIFY(s)
+#endif
+
+#ifndef PURE
+#define PURE __attribute__((pure))
+#endif
+
+#define FORCE_INLINE __attribute__((always_inline))
+#define FORCE_NO_INLINE __attribute__((noinline))
+#define NO_OPTIMIZE __attribute__((optimize("-O0")))
+
+#ifndef C_ASSERT
+#ifdef __cplusplus
+#define C_ASSERT(e) typedef char __C_ASSERT__[(e) ? 1 : -1]
+#else
+#if __clang__
+#define C_ASSERT(e) _Static_assert((e), #e)
+#else
+#define C_ASSERT(e) (extern char(*ct_assert(void))[sizeof(char[1 - 2 * !(e)])])
+#endif
+#endif
+#endif
+
+#ifndef CONTAINER_OF
+#define CONTAINER_OF(ptr, type, member)\
+	((type *) ((char *) (ptr) - (char *) &((type *) 0)->member))
+#endif
+
+#if !defined(__cplusplus) && !defined(offsetof)
+#define offsetof(st, m) __builtin_offsetof(st, m)
+#endif
+
+//Wrap 'static' in #define to allow override for symbol/function globalization
+#if !defined(UT)
+#define PRIVATE static
+#define STATIC static
+#define INLINE inline
+
+#else
+#define PRIVATE
+#define STATIC
+#define INLINE
+#endif
+
+#define C_CAST(type, value) ((type)(value))
+
+#if defined(_MSC_VER)
+#define PACKED
+#else
+#define PACKED __attribute__((__packed__))
+#endif
+
+/*! Returns protected region size in kB. */
+/*!
+ *\param value protected region size is calculated form value (e.g. IMR_PROTECTION_128_KB_SIZE)
+ */
+#define PROTECTED_REGION_SIZE(value) (BIT(value) * 0x1000)
+
+#define DEFINE_UUID_(name, l1, s1, s2, b1, b2, b3, b4, b5, b6, b7, b8) \
+static const uint32_t name[] = \
+{ \
+	l1, \
+	s2 << 16 | s1, \
+	b4 << 24 | b3 << 16 | b2 << 8 | b1, \
+	b8 << 24 | b7 << 16 | b6 << 8 | b5, \
+}
+#define DEFINE_UUID(name, ...) DEFINE_UUID_(name, __VA_ARGS__)
+#define DEFINE_UUID_INLINE_(l1, s1, s2, b1, b2, b3, b4, b5, b6, b7, b8) \
+	(l1, \
+	s2 << 16 | s1, \
+	b4 << 24 | b3 << 16 | b2 << 8 | b1, \
+	b8 << 24 | b7 << 16 | b6 << 8 | b5)
+#define DEFINE_UUID_INLINE(...) DEFINE_UUID_INLINE_(__VA_ARGS__)
+
+#if defined(__KLOCWORK__)
+#include "misc/klocwork_overrides.h"
+#endif
+
+#ifndef __syscall
+#if !(defined(ULP_FW) && defined(UT)) /* if not UT for ULP, where call mockups are used*/
+#define __syscall(...) (static inline)
+#else
+#define __syscall(...)
+#endif
+#endif
+
+#endif // SYNTAX_SUGAR_H
```

Comment:
hmm, this file has a lot of duplicated high level / core macros, we need to reuse.